### PR TITLE
noImplictAny: No errors left

### LINKED ISSUE #14714: Typescript: noImplicitAny: true
Would be nice to move to set this to true. 

Problem is that currently enabling this typecheck results in:

Found 7228 errors.

### PR DESCRIPTION
**Pull request by tskarhed**
Closes #14714

### CODE REVIEW

File: public/app/core/utils/connectWithReduxStore.tsx [hunk 1]
```diff
@@ -2,10 +2,10 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { store } from '../../store/store';
 
-export function connectWithStore(WrappedComponent, ...args) {
+export function connectWithStore(WrappedComponent: any, ...args: any[]) {
```
Comment (dehrax) [inline public/app/core/utils/connectWithReduxStore.tsx]: Is there a better type for `WrappedComponent`? Any extenstion of `Component` or something similar?

Comment (torkelo) [reply public/app/core/utils/connectWithReduxStore.tsx]: probably but not sure it's worth it

File: public/app/core/utils/emitter.ts [hunk 2]
```diff
@@ -7,11 +7,11 @@ export class Emitter {
     this.emitter = new EventEmitter();
   }
 
-  emit(name, data?) {
+  emit(name: string, data?: any) {
     this.emitter.emit(name, data);
   }
 
-  on(name, handler, scope?) {
+  on(name: string, handler: any, scope?: any) {
```
Comment (torkelo) [inline public/app/core/utils/emitter.ts]: handler should be a function with signature (payload?: any) => void

File: public/app/core/utils/emitter.ts [hunk 3]
```diff
@@ -22,11 +22,11 @@ export class Emitter {
     }
   }
 
-  removeAllListeners(evt?) {
+  removeAllListeners(evt?: any) {
```
Comment (torkelo) [inline public/app/core/utils/emitter.ts]: maybe should be string?

File: public/app/core/utils/emitter.ts [hunk 4]
```diff
@@ -22,11 +22,11 @@ export class Emitter {
...
     this.emitter.removeAllListeners(evt);
   }
 
-  off(name, handler) {
+  off(name: any, handler: any) {
```
Comment (torkelo) [inline public/app/core/utils/emitter.ts]: string and handler type

File: public/app/core/utils/file_export.ts [hunk 5]
```diff
@@ -167,28 +172,28 @@ function mergeSeriesByTime(seriesList) {
...
 }
 
-export function exportSeriesListToCsvColumns(seriesList, options: Partial<SeriesListToCsvColumnsOptions>) {
+export function exportSeriesListToCsvColumns(seriesList: SeriesList, options: Partial<SeriesListToCsvColumnsOptions>) {
   const text = convertSeriesListToCsvColumns(seriesList, options);
   saveSaveBlob(text, EXPORT_FILENAME);
 }
 
-export function convertTableDataToCsv(table, excel = false) {
+export function convertTableDataToCsv(table: any, excel = false) {
```
Comment (torkelo) [inline public/app/core/utils/file_export.ts]: TableData maybe better or TableModel

File: public/app/core/utils/model_utils.ts [hunk 6]
```diff
@@ -1,4 +1,9 @@
-export function assignModelProperties(target, source, defaults, removeDefaults?) {
+export function assignModelProperties(
+  target: any,
+  source: any,
+  defaults: { [x: string]: any; propB?: number; propC?: number; hasOwnProperty?: any },
```
Comment (torkelo) [inline public/app/core/utils/model_utils.ts]: this looks a bit strange.

File: public/app/features/annotations/editor_ctrl.ts [hunk 7]
```diff
@@ -37,13 +38,13 @@ export class AnnotationsEditorCtrl {
     this.onColorChange = this.onColorChange.bind(this);
   }
 
-  datasourceChanged() {
+  async datasourceChanged() {
```
Comment (torkelo) [inline public/app/features/annotations/editor_ctrl.ts]: do you need async when there is no await in this function?

File: public/app/features/annotations/editor_ctrl.ts [hunk 8]
```diff
@@ -73,7 +74,7 @@ export class AnnotationsEditorCtrl {
     this.mode = 'list';
   }
 
-  move(index, dir) {
+  move(index: number, dir: any) {
```
Comment (torkelo) [inline public/app/features/annotations/editor_ctrl.ts]: dir should be number?

File: public/app/features/dashboard/components/AddPanelWidget/AddPanelWidget.tsx [hunk 9]
```diff
@@ -93,7 +93,7 @@ export class AddPanelWidget extends React.Component<Props, State> {
     reduxStore.dispatch(updateLocation(location));
   };
 
-  onPasteCopiedPanel = panelPluginInfo => {
+  onPasteCopiedPanel = (panelPluginInfo: { id: any; defaults: { gridPos: { w: any; h: any }; title: any } }) => {
```
Comment (torkelo) [inline public/app/features/dashboard/components/AddPanelWidget/AddPanelWidget.tsx]: extract this type in same file instead, a bit too much inline

File: public/app/features/dashboard/components/DashLinks/DashLinksContainerCtrl.ts [hunk 10]
```diff
@@ -138,7 +149,7 @@ export class DashLinksContainerCtrl {
       });
     }
 
-    $scope.searchDashboards = (link, limit) => {
+    $scope.searchDashboards = (link: { tags: any; target: string; keepTime: any; includeVars: any }, limit: any) => {
```
Comment (torkelo) [inline public/app/features/dashboard/components/DashLinks/DashLinksContainerCtrl.ts]: can you define a DashboardLink type for this?

**Review by torkelo — APPROVED** [review]
Great work! just some minor improvements then this is done!

**Review by torkelo — APPROVED** [review]

**Pull request merged**